File: java/rest-simple/old_structure/service/StudentService.java
package com.example.restsimple.service;

import com.example.restsimple.model.Student;
import com.example.restsimple.repository.UpdateStudent;

import java.util.List;
import java.util.Optional;

public interface StudentService {
    List<Student> getAllStudents();

    Student createStudent(Student student);

    Optional<Student> updateStudent(String id, UpdateStudent updatedStudent);

    Optional<Student> deleteStudent(String id);

    Optional<Student> findStudentById(String id);
}

File: kotlin/rest-simple-exercise/src/main/kotlin/com/example/restsimple/service/StudentService.kt
package com.example.restsimple.service

import com.example.restsimple.model.Student
import com.example.restsimple.repository.UpdateStudent
import java.util.*

interface StudentService {
    val allStudents: MutableList<Student?>
    fun createStudent(student: Student): Student?
    fun updateStudent(id: String?, updatedStudent: UpdateStudent): Optional<out Student?>
    fun deleteStudent(id: String?): Optional<out Student?>
    fun findStudentById(id: String?): Optional<Student>
}
